frontend/src/metabase/browse/components/BrowseModels.tsx

```diff
@@ -0,0 +1,192 @@
+
+import type { Card, Collection, SearchResult } from "metabase-types/api";
+import * as Urls from "metabase/lib/urls";
+
+import Link from "metabase/core/components/Link";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+
+import type { useSearchListQuery } from "metabase/common/hooks";
+
+import { Box, Group, Icon, Text, Title } from "metabase/ui";
+import { sortModels } from "metabase/browse/utils";
+import NoResults from "assets/img/no_results.svg";
+import { useSelector } from "metabase/lib/redux";
+import { getLocale } from "metabase/setup/selectors";
+import { CenteredEmptyState } from "./BrowseApp.styled";
+import {
+  CollectionHeaderContainer,
+  CollectionHeaderLink,
+  GridContainer,
+  ModelCard,
+  MultilineEllipsified,
+} from "./BrowseModels.styled";
+import { LastEdited } from "./LastEdited";
+
+const emptyArray: SearchResult[] = [];
+
+export const groupModels = (
+  models: SearchResult[],
+  locale: string | undefined,
+) => {
+  const groupedModels = _.groupBy(models, model => model.collection.id);
+  let collections = models.map(model => model.collection);
+  collections = _.uniq(collections, collection => collection.id) || [];
+  collections.sort((a, b) => a.name?.localeCompare(b.name, locale));
+  return { groupedModels, collections };
+};
+
+export const BrowseModels = ({
+  modelsResult,
+}: {
+  modelsResult: ReturnType<typeof useSearchListQuery<SearchResult>>;
+}) => {
+  const { data: models = emptyArray, error, isLoading } = modelsResult;
+  const locale = useSelector(getLocale);
+  const { collections, groupedModels } = groupModels(models, locale?.code);
+
+  if (error || isLoading) {
+    return (
+      <LoadingAndErrorWrapper
+        error={error}
+        loading={isLoading}
+        style={{ display: "flex", flex: 1 }}
+      />
+    );
+  }
+
+  if (models.length) {
+    return (
+      <GridContainer role="grid">
+        {collections.map((collection, index) => (
+          <ModelGroup
+            index={index}
+            models={groupedModels[collection.id]}
+            key={`modelgroup-${index}`}
+          />
+        ))}
+      </GridContainer>
+    );
+  }
+
+  return (
+    <CenteredEmptyState
+      title={<Box mb=".5rem">{t`No models here yet`}</Box>}
+      message={
+        <Box maw="24rem">{t`Models help curate data to make it easier to find answers to questions all in one place.`}</Box>
+      }
+      illustrationElement={
+        <Box mb=".5rem">
+          <img src={NoResults} />
+        </Box>
+      }
+    />
+  );
+};
+
+const ModelGroup = ({
+  models,
+  index,
+}: {
+  models: SearchResult[];
+  index: number;
+}) => {
+  const sortedModels = models.sort(sortModels);
+  const collection = models[0].collection;
+
+  /** This id is used by aria-labelledby */
+  const collectionHtmlId = `collection-${collection?.id ?? `index-${index}`}`;
+
+  return (
+    <>
+      <CollectionHeader
+        collection={models[0].collection}
```

Comment:
I think you probably just want to use `collection` that you extracted already here